In [1]:
import os
import pandas as pd
import numpy as np


In [2]:
file_path = "Data/"

In [3]:
df = pd.read_excel(f'{file_path}2019 County Health Rankings Data - v3.xls', header=1, sheet_name=5)
df.head()

FIPS    State   County  Life Expectancy  95% CI - Low  95% CI - High  \
0  1001  Alabama  Autauga        76.330589     75.554117      77.107061   
1  1003  Alabama  Baldwin        78.599498     78.208357      78.990639   
2  1005  Alabama  Barbour        75.779457     74.639355      76.919559   
3  1007  Alabama     Bibb        73.928271     72.624611      75.231931   
4  1009  Alabama   Blount        74.597767     73.810999      75.384535   

   Life Expectancy (Black)  Life Expectancy (Hispanic)  \
0                74.600628                         NaN   
1                75.628919                   98.039893   
2                74.240228                         NaN   
3                72.294686                         NaN   
4                      NaN                   78.577094   

   Life Expectancy (White)  # Deaths  ...  % Hispanic  # Non-Hispanic White  \
0                76.337219     815.0  ...    2.857452                 41336   
1                78.537003    2827.0  ...    4.550200                176582   
2                77.568384     451.0  ...    4.206569                 11613   
3                74.109510     445.0  ...    2.638080                 16842   
4                74.098247    1050.0  ...    9.565097                 50439   

   % Non-Hispanic White  # Not Proficient in English  \
0             74.473912                          430   
1             83.047388                          872   
2             45.955679                          297   
3             74.298571                           84   
4             86.944306                         1013   

   % Not Proficient in English  95% CI - Low.16  95% CI - High.16   % Female  \
0                     0.828548         0.422472          1.234624  51.342246   
1                     0.454512         0.282144          0.626880  51.452772   
2                     1.199273         0.580536          1.818011  47.229917   
3                     0.394348         0.000000          0.933361  46.453150   
4                     1.870004         1.364180          2.375829  50.688639   

   # Rural    % Rural  
0  22921.0  42.002162  
1  77060.0  42.279099  
2  18613.0  67.789635  
3  15663.0  68.352607  
4  51562.0  89.951502  

[5 rows x 111 columns]

In [4]:
for col in df.columns:
    print(col)

FIPS
State
County
Life Expectancy
95% CI - Low
95% CI - High
Life Expectancy (Black)
Life Expectancy (Hispanic)
Life Expectancy (White)
# Deaths
Age-Adjusted Mortality
95% CI - Low.1
95% CI - High.1
Age-Adjusted Mortality (Black)
Age-Adjusted Mortality (Hispanic)
Age-Adjusted Mortality (White)
# Deaths.1
Child Mortality Rate
95% CI - Low.2
95% CI - High.2
Child Mortality Rate (Black)
Child Mortality Rate (Hispanic)
Child Mortality Rate (White)
# Deaths.2
Infant Mortality Rate
95% CI - Low.3
95% CI - High.3
Infant Mortality Rate (Black)
Infant Mortality Rate (Hispanic)
Infant Mortality Rate (White)
% Frequent Physical Distress
95% CI - Low.4
95% CI - High.4
% Frequent Mental Distress
95% CI - Low.5
95% CI - High.5
% Diabetic
95% CI - Low.6
95% CI - High.6
# HIV Cases
HIV Prevalence Rate
# Food Insecure
% Food Insecure
# Limited Access
% Limited Access
# Drug Overdose Deaths
Drug Overdose Mortality Rate
# Motor Vehicle Deaths
MV Mortality Rate
95% CI - Low.7
95% CI - High.7
% Insufficien

In [5]:
df_demographics = pd.DataFrame(data={"FIPS":df['FIPS'],
                                     "State":df['State'],
                                     "County":df['County'],
                                     "Population":df['Population'],
                                     "%<18_Yrs_Old":df['% < 18'].map("{:.2f}".format),
                                     "%65_Yrs_Old_and_over": df['% 65 and over'].map("{:.2f}".format),
                                     "Income($)": df['Household Income'].map("${:.2f}".format),
                                     "African_American(Count)": df['# African American'],
                                     "African_American(%)": df['% African American'].map("{:.2f}".format),
                                     "American_Indian/Alaskan_Native(Count)": df['# American Indian/Alaskan Native'],
                                     "American_Indian/Alaskan_Native(%)": df['% American Indian/Alaskan Native'].map("{:.2f}".format),
                                     "Asian(Count)": df['# Asian'],
                                     "Asian(%)": df['% Asian'].map("{:.2f}".format),
                                     "Native_Hawaiian/Other_Pacific_Islander(Count)":df['# Native Hawaiian/Other Pacific Islander'], 
                                     "Native_Hawaiian/Other_Pacific_Islander(%)": df['% Native Hawaiian/Other Pacific Islander'].map("{:.2f}".format), 
                                     "Hispanic(Count)": df['# Hispanic'], 
                                     "Hispanic(%)": df['% Hispanic'].map("{:.2f}".format), 
                                     "Non-Hispanic_White_(Count)":df['# Non-Hispanic White'], 
                                     "Non-Hispanic_White(%)": df['% Non-Hispanic White'].map("{:.2f}".format), 
                                     "Female(%)": df['% Female'].map("{:.2f}".format), 
                                     "Rural(Count)": df['# Rural'], 
                                     "Rural(%)": df['% Rural'].map("{:.2f}".format),
                                     "Life_Expectancy": df['Life Expectancy'].map("{:.2f}".format),
                                     "Deaths(Count)": df['# Deaths']
                                    })
print(df_demographics.shape)
df_demographics.head(15)

(3142, 24)


FIPS    State    County  Population % < 18 Yrs Old % 65 Yrs Old and over  \
0   1001  Alabama   Autauga       55504          23.94                 15.12   
1   1003  Alabama   Baldwin      212628          21.85                 19.95   
2   1005  Alabama   Barbour       25270          20.76                 18.82   
3   1007  Alabama      Bibb       22668          20.61                 16.02   
4   1009  Alabama    Blount       58013          23.35                 17.84   
5   1011  Alabama   Bullock       10309          20.71                 15.82   
6   1013  Alabama    Butler       19825          22.73                 19.56   
7   1015  Alabama   Calhoun      114728          21.79                 17.31   
8   1017  Alabama  Chambers       33713          21.14                 19.08   
9   1019  Alabama  Cherokee       25857          19.71                 22.62   
10  1021  Alabama   Chilton       44067          23.88                 16.36   
11  1023  Alabama   Choctaw       12945          20.09                 22.74   
12  1025  Alabama    Clarke       24083          21.88                 19.28   
13  1027  Alabama      Clay       13367          20.57                 20.24   
14  1029  Alabama  Cleburne       14900          22.95                 19.11   

   Income ($)  African American (Count) African American (%)  \
0   $58343.00                     10687                19.25   
1   $56607.00                     19037                 8.95   
2   $32490.00                     12115                47.94   
3   $45795.00                      4864                21.46   
4   $48253.00                       847                 1.46   
5   $29113.00                      7130                69.16   
6   $36842.00                      8845                44.62   
7   $45937.00                     23852                20.79   
8   $36883.00                     13313                39.49   
9   $44842.00                      1090                 4.22   
10  $46405.00                      4442                10.08   
11  $36711.00                      5389                41.63   
12  $36719.00                     10634                44.16   
13  $39642.00                      1849                13.83   
14  $45480.00                       459                 3.08   

    American Indian/Alaskan Native (Count)  ...  \
0                                      264  ...   
1                                     1650  ...   
2                                      165  ...   
3                                       97  ...   
4                                      367  ...   
5                                       87  ...   
6                                       75  ...   
7                                      607  ...   
8                                      100  ...   
9                                      163  ...   
10                                     241  ...   
11                                      23  ...   
12                                     119  ...   
13                                      77  ...   
14                                      69  ...   

   Native Hawaiian/Other Pacific Islander (%)  Hispanic (Count) Hispanic (%)  \
0                                        0.10              1586         2.86   
1                                        0.07              9675         4.55   
2                                        0.19              1063         4.21   
3                                        0.11               598         2.64   
4                                        0.12              5549         9.57   
5                                        0.78               850         8.25   
6                                        0.05               285         1.44   
7                                        0.11              4302         3.75   
8                                        0.04               822         2.44   
9                                        0.03               425         1.64   
1

In [6]:
df_demographics.dtypes

FIPS                                                int64
State                                              object
County                                             object
Population                                          int64
% < 18 Yrs Old                                     object
% 65 Yrs Old and over                              object
Income ($)                                         object
African American (Count)                            int64
African American (%)                               object
American Indian/Alaskan Native (Count)              int64
American Indian/Alaskan Native(%)                  object
Asian (Count)                                       int64
Asian (%)                                          object
Native Hawaiian/Other Pacific Islander (Count)      int64
Native Hawaiian/Other Pacific Islander (%)         object
Hispanic (Count)                                    int64
Hispanic (%)                                       object
Non-Hispanic W

In [7]:
type(df_demographics['FIPS'])

pandas.core.series.Series

In [8]:
lyme_df = pd.read_csv(f'{file_path}LymeCaseCountsbyCounty_FIPS.csv', encoding='latin-1')
lyme_df.head(20)

Ctyname Ctyname_Updated         Stname  \
0          Alabama         Alabama        Alabama   
1           Alaska          Alaska         Alaska   
2          Arizona         Arizona        Arizona   
3       California      California     California   
4         Colorado        Colorado       Colorado   
5      Connecticut     Connecticut    Connecticut   
6         Delaware        Delaware       Delaware   
7          Florida         Florida        Florida   
8          Georgia         Georgia        Georgia   
9         Illinois        Illinois       Illinois   
10         Indiana         Indiana        Indiana   
11          Kansas          Kansas         Kansas   
12        Kentucky        Kentucky       Kentucky   
13  LaSalle Parish         LaSalle      Louisiana   
14        Lousiana        Lousiana      Louisiana   
15           Maine           Maine          Maine   
16        Maryland        Maryland       Maryland   
17   Massachusetts   Massachusetts  Massachusetts   
18        Michigan        Michigan       Michigan   
19       Minnesota       Minnesota      Minnesota   

                Unique Identifier   FIPS  STCODE  CTYCODE  Cases2000  \
0               Alabama | Alabama  99901       1      999          0   
1                 Alaska | Alaska  99902       2      999          0   
2               Arizona | Arizona  99903       4      999          0   
3         California | California  99904       6      999          0   
4             Colorado | Colorado  99905       8      999          0   
5       Connecticut | Connecticut  99906       9      999        111   
6             Delaware | Delaware  99907      10      999          0   
7               Florida | Florida  99909      12      999          0   
8               Georgia | Georgia  99910      13      999          0   
9             Illinois | Illinois  99911      17      999          0   
10              Indiana | Indiana  99912      18      999          0   
11                Kansas | Kansas  99913      20      999          0   
12            Kentucky | Kentucky  99914      21      999          0   
13            Louisiana | LaSalle  99915      22       59          0   
14           Louisiana | Lousiana  99916      22      999          0   
15                  Maine | Maine  99917      23      999          1   
16            Maryland | Maryland  99918      24      999          0   
17  Massachusetts | Massachusetts  99919      25      999          7   
18            Michigan | Michigan  99920      26      999          0   
19          Minnesota | Minnesota  99921      27      999          3   

    Cases2001  Cases2002  ...  Cases2010  Cases2011  Cases2012  Cases2013  \
0           0          0  ...          0          0          0          0   
1           0          0  ...          0          0          1          1   
2           0          0  ...          0          0          0          0   
3           0          0  ...          0          0          0          0   
4           0          0  ...          0          0          0          0   
5         122        135  ...       1448       1432       1402        597   
6           0          0  ...         37         34         38         29   
7           0          0  ...          0          0          0          0   
8           0          0  ...          0          1          0          0   
9           0          0  ...          0          0          0          0   
10          0          0  ...         12         10          7         19   
11          0          0  ...          0          0          0          0   
12          0          0  ...          0          0          9         24   
13          0          0  ...          0          0          0          0   
14          1          0  ...          0          0          0          0   
15          0          3  ...          0          0          0          0   
16          0          0  ...          0          1          3          0  

In [9]:
df_lyme_2019 = lyme_df.drop(columns=['STCODE', 'CTYCODE', 'Unique Identifier','Ctyname', 'Cases2000', 'Cases2001',
       'Cases2002', 'Cases2003', 'Cases2004', 'Cases2005', 'Cases2006',
       'Cases2007', 'Cases2008', 'Cases2009', 'Cases2010', 'Cases2011',
       'Cases2012', 'Cases2013', 'Cases2014', 'Cases2015', 'Cases2016','Cases2017','Cases2018'])
df_lyme_2019 = df_lyme_2019.rename(columns={'Ctyname_Updated':'County','Stname':'State','Cases2019':'Lyme_Disease_Incidence_Reported'})
print(df_lyme_2019.shape)
df_lyme_2019.head()

(3193, 4)


County       State   FIPS  Lyme Disease Incidence Reported
0     Alabama     Alabama  99901                                0
1      Alaska      Alaska  99902                                0
2     Arizona     Arizona  99903                                0
3  California  California  99904                                0
4    Colorado    Colorado  99905                                0

In [10]:
df_lyme_2019 = df_lyme_2019.sort_values('FIPS', ascending=False)
print(df_lyme_2019.shape)
df_lyme_2019.head(10)

(3193, 4)


County           State   FIPS  Lyme Disease Incidence Reported
43         Wyoming         Wyoming  99945                                3
42       Wisconsin       Wisconsin  99944                                0
41   West Virginia   West Virginia  99943                              256
40      Washington      Washington  99942                                0
39        Virginia        Virginia  99941                              124
38         Vermont         Vermont  99940                                0
37           Texas           Texas  99939                               20
36       Tennessee       Tennessee  99938                               14
35    South Dakota    South Dakota  99937                                0
34  South Carolina  South Carolina  99936                                0

In [11]:
df_lyme_2019.dtypes

County                             object
State                              object
FIPS                                int64
Lyme Disease Incidence Reported     int64
dtype: object

In [12]:
df_lyme_2019.to_csv('Data/Lyme2019DF.csv', index=False)

In [13]:
df_merged = df_lyme_2019.merge(df_demographics, how ='right')


print(df_merged.shape)
df_merged.head()

(3150, 25)


County    State  FIPS  Lyme Disease Incidence Reported  Population  \
0  Autauga  Alabama  1001                              1.0       55504   
1  Baldwin  Alabama  1003                              0.0      212628   
2  Barbour  Alabama  1005                              1.0       25270   
3     Bibb  Alabama  1007                              0.0       22668   
4   Blount  Alabama  1009                              0.0       58013   

  % < 18 Yrs Old % 65 Yrs Old and over Income ($)  African American (Count)  \
0          23.94                 15.12  $58343.00                     10687   
1          21.85                 19.95  $56607.00                     19037   
2          20.76                 18.82  $32490.00                     12115   
3          20.61                 16.02  $45795.00                      4864   
4          23.35                 17.84  $48253.00                       847   

  African American (%)  ...  Native Hawaiian/Other Pacific Islander (%)  \
0                19.25  ...                                        0.10   
1                 8.95  ...                                        0.07   
2                47.94  ...                                        0.19   
3                21.46  ...                                        0.11   
4                 1.46  ...                                        0.12   

  Hispanic (Count)  Hispanic (%) Non-Hispanic White (Count)  \
0             1586          2.86                      41336   
1             9675          4.55                     176582   
2             1063          4.21                      11613   
3              598          2.64                      16842   
4             5549          9.57                      50439   

   Non-Hispanic White (%) Female (%)  Rural (Count) Rural (%)  \
0                   74.47      51.34        22921.0     42.00   
1                   83.05      51.45        77060.0     42.28   
2                   45.96      47.23        18613.0     67.79   
3                   74.30      46.45        15663.0     68.35   
4                   86.94      50.69        51562.0     89.95   

   Life Expectancy Deaths (Count)  
0            76.33          815.0  
1            78.60         2827.0  
2            75.78          451.0  
3            73.93          445.0  
4            74.60         1050.0  

[5 rows x 25 columns]

In [14]:
df_merged.dtypes

County                                             object
State                                              object
FIPS                                                int64
Lyme Disease Incidence Reported                   float64
Population                                          int64
% < 18 Yrs Old                                     object
% 65 Yrs Old and over                              object
Income ($)                                         object
African American (Count)                            int64
African American (%)                               object
American Indian/Alaskan Native (Count)              int64
American Indian/Alaskan Native(%)                  object
Asian (Count)                                       int64
Asian (%)                                          object
Native Hawaiian/Other Pacific Islander (Count)      int64
Native Hawaiian/Other Pacific Islander (%)         object
Hispanic (Count)                                    int64
Hispanic (%)  

In [15]:
# Check for null values
for col in df_merged:
    print(f'{col} has {df_merged[col].isnull().sum()} nulls')

County has 0 nulls
State has 0 nulls
FIPS has 0 nulls
Lyme Disease Incidence Reported has 42 nulls
Population has 0 nulls
% < 18 Yrs Old has 0 nulls
% 65 Yrs Old and over has 0 nulls
Income ($) has 0 nulls
African American (Count) has 0 nulls
African American (%) has 0 nulls
American Indian/Alaskan Native (Count) has 0 nulls
American Indian/Alaskan Native(%) has 0 nulls
Asian (Count) has 0 nulls
Asian (%) has 0 nulls
Native Hawaiian/Other Pacific Islander (Count) has 0 nulls
Native Hawaiian/Other Pacific Islander (%) has 0 nulls
Hispanic (Count) has 0 nulls
Hispanic (%) has 0 nulls
Non-Hispanic White (Count) has 0 nulls
Non-Hispanic White (%) has 0 nulls
Female (%) has 0 nulls
Rural (Count) has 7 nulls
Rural (%) has 0 nulls
Life Expectancy has 0 nulls
Deaths (Count) has 61 nulls


In [16]:
df_merged.loc[df_merged['FIPS'] == 99940]

Empty DataFrame
Columns: [County, State, FIPS, Lyme Disease Incidence Reported, Population, % < 18 Yrs Old, % 65 Yrs Old and over, Income ($), African American (Count), African American (%), American Indian/Alaskan Native (Count), American Indian/Alaskan Native(%), Asian (Count), Asian (%), Native Hawaiian/Other Pacific Islander (Count), Native Hawaiian/Other Pacific Islander (%), Hispanic (Count), Hispanic (%), Non-Hispanic White (Count), Non-Hispanic White (%), Female (%), Rural (Count), Rural (%), Life Expectancy, Deaths (Count)]
Index: []

[0 rows x 25 columns]

In [17]:
# Drop rows with NaNs
df_merged = df_merged.dropna()

In [18]:
print(f'Are there any null values? {df_merged.isnull().values.any()}')
print(df_merged.shape)
df_merged.head(20)

Are there any null values? False
(3041, 25)


County    State  FIPS  Lyme Disease Incidence Reported  Population  \
0     Autauga  Alabama  1001                              1.0       55504   
1     Baldwin  Alabama  1003                              0.0      212628   
2     Barbour  Alabama  1005                              1.0       25270   
3        Bibb  Alabama  1007                              0.0       22668   
4      Blount  Alabama  1009                              0.0       58013   
5     Bullock  Alabama  1011                              0.0       10309   
6      Butler  Alabama  1013                              0.0       19825   
7     Calhoun  Alabama  1015                              3.0      114728   
8    Chambers  Alabama  1017                              2.0       33713   
9    Cherokee  Alabama  1019                              0.0       25857   
10    Chilton  Alabama  1021                              1.0       44067   
11    Choctaw  Alabama  1023                              0.0       12945   
12     Clarke  Alabama  1025                              0.0       24083   
13       Clay  Alabama  1027                              0.0       13367   
14   Cleburne  Alabama  1029                              1.0       14900   
15     Coffee  Alabama  1031                              0.0       51874   
16    Colbert  Alabama  1033                              2.0       54500   
17    Conecuh  Alabama  1035                              0.0       12469   
18      Coosa  Alabama  1037                              1.0       10754   
19  Covington  Alabama  1039                              0.0       37092   

   % < 18 Yrs Old % 65 Yrs Old and over Income ($)  African American (Count)  \
0           23.94                 15.12  $58343.00                     10687   
1           21.85                 19.95  $56607.00                     19037   
2           20.76                 18.82  $32490.00                     12115   
3           20.61                 16.02  $45795.00                      4864   
4           23.35                 17.84  $48253.00                       847   
5           20.71                 15.82  $29113.00                      7130   
6           22.73                 19.56  $36842.00                      8845   
7           21.79                 17.31  $45937.00                     23852   
8           21.14                 19.08  $36883.00                     13313   
9           19.71                 22.62  $44842.00                      1090   
10          23.88                 16.36  $46405.00                      4442   
11          20.09                 22.74  $36711.00                      5389   
12          21.88                 19.28  $36719.00                     10634   
13          20.57                 20.24  $39642.00                      1849   
14          22.95                 19.11  $45480.00                       459   
15          23.72                 16.57  $51998.00                      8995   
16          21.01                 19.60  $46100.00                      8763   
17          21.03                 21.92  $30796.00                      5758   
18          16.40                 22.76  $42442.00                      3245   
19          21.83                 20.66  $39093.00                      4606   

   African American (%)  ...  Native Hawaiian/Other Pacific Islander (%)  \
0                 19.25  ...                                        0.10   
1                  8.95  ...                                        0.07   
2                 47.94  ...                                        0.19   
3                 21.46  ...                                        0.11   
4                  1.46  ...                                        0.12   
5                 69.16  ...                                        0.78   
6                 44.62  ...                                        0.05   
7                 20.79  ...                                        0.11   
8                 39.49 

In [19]:
# Check for the incidents of lyme disease reported by county
cases_df = df_merged.groupby(['State','County'])
cases_df.sum().head(15)

FIPS  Lyme Disease Incidence Reported  Population  \
State   County                                                        
Alabama Autauga   1001                              1.0       55504   
        Baldwin   1003                              0.0      212628   
        Barbour   1005                              1.0       25270   
        Bibb      1007                              0.0       22668   
        Blount    1009                              0.0       58013   
        Bullock   1011                              0.0       10309   
        Butler    1013                              0.0       19825   
        Calhoun   1015                              3.0      114728   
        Chambers  1017                              2.0       33713   
        Cherokee  1019                              0.0       25857   
        Chilton   1021                              1.0       44067   
        Choctaw   1023                              0.0       12945   
        Clarke    1025                              0.0       24083   
        Clay      1027                              0.0       13367   
        Cleburne  1029                              1.0       14900   

                  African American (Count)  \
State   County                               
Alabama Autauga                      10687   
        Baldwin                      19037   
        Barbour                      12115   
        Bibb                          4864   
        Blount                         847   
        Bullock                       7130   
        Butler                        8845   
        Calhoun                      23852   
        Chambers                     13313   
        Cherokee                      1090   
        Chilton                       4442   
        Choctaw                       5389   
        Clarke                       10634   
        Clay                          1849   
        Cleburne                       459   

                  American Indian/Alaskan Native (Count)  Asian (Count)  \
State   County                                                            
Alabama Autauga                                      264            710   
        Baldwin                                     1650           2459   
        Barbour                                      165            117   
        Bibb                                          97             50   
        Blount                                       367            175   
        Bullock                                       87             20   
        Butler                                        75            204   
        Calhoun                                      607           1137   
        Chambers                                     100            421   
        Cherokee                                     163             77   
        Chilton                                      241            220   
        Choctaw                                       23             19   
        Clarke                                       119            121   
        Clay                                          77             33   
        Cleburne                                      69             30   

                  Native Hawaiian/Other Pacific Islander (Count)  \
State   County                                                     
Alabama Autauga                                               58   
        Baldwin                                              146   
        Barbour                                               47   
        Bibb                                                  26   
        Blount                                                68   
        Bullock                                               80   
        Butler                                                10   
        Calhoun                                              129   
        Chambers                                              15   
   

In [20]:
# Export Merged DF as csv
df_merged.to_csv('Data/Merged_Demographic_Data.csv', index=False)

### Join methon 'right' was decided to include a majority of the county demographic info and ommit the '999' county codes rows from the lyme incidence file. This results in less NaNs and more overall data for analysis. However, the tradeoff being there will be less Lyme disease incidence repoted data 